### PR TITLE
feat: add notebooks sidebar

### LINKED ISSUE #1337: Notebooks: I can see all the options for a panel in a side menu
When: I have click on a notebook cell
Then: I see the notebook cell highlighted with an active stroke
And: A side menu with options that are specific to that panel

### Design Notes

Work in Progress:
https://www.figma.com/file/8S8IH8mimYIwanZ6keIQ6U/Notebooks-Prototype?node-id=2236%3A3

Spreadsheet of panel content:
https://docs.google.com/spreadsheets/d/1pYjS8CJhnFoNxtNzREOoSDQrhJAYyPdOEYsXW7L3OYA/edit#gid=0

Dropdown component for reference: https://influxdata.github.io/clockface/?path=/story/components-dropdowns-family--dropdownmenu

Icons used in Panel Menu: 
`Export`
`Cog Solid`
`Eye Open`

Panel selection interaction:
- Any click event in the child of the DOM should bubble up to the parent
- A child click event should be able to make the entire panel active and trigger the panel menu
- We would need to remove any explicit no click event Javascript overrides if there are any, such as stop propagation or prevent default

### LINKED ISSUE #1338: Notebooks: I can see functions for script panel in panel submenu
Given: I have a flux script panel
When: I click on Functions in the panel menu
Then: I see `Function` as a submenu

And: When I click back
Then: I go back to the panel menu

#### Design Notes

https://www.figma.com/file/8S8IH8mimYIwanZ6keIQ6U/Notebooks-Prototype?node-id=2251%3A10933

### LINKED ISSUE #1339: Notebooks: I can see dashboard setting in a submenu
Given: I have a notebook with a visualization panel
When: I click `customize` on the panel menu
Then: I can see the options to customize my visualization type in a submenu

When: I see the customization submenu
Then: I fixed height relative to the panel (~600-900 px)
And: I can scroll to see all options

When: I click back
Then: I got to the panel menu

#### Design Notes

Prototype
https://www.figma.com/proto/8S8IH8mimYIwanZ6keIQ6U/Notebooks-Prototype?node-id=2715%3A3&scaling=min-zoom&page-id=1915%3A0

Working File
https://www.figma.com/file/8S8IH8mimYIwanZ6keIQ6U/Notebooks-Prototype?node-id=2502%3A12290

### LINKED ISSUE #1341: Notebooks: Generate and Edit Flux from a Panel
As a notebooks advanced user, I want to see the flux query generated by a panel so that I can extend or write additional custom flux. 

## Scenario 1: Default Behavior
**Given:** I have a notebook with at least one panel
**When:** I click `convert to flux` in the panel menu
**Then:** the notebook panel is deleted
**And:** Replaced with a raw flux panel that displays the source query

## Scenario 2: Visualization Panels
**If:** there is an aggregate or window period that's different from the source query
**Then:**  we should show the visualization query and the source query, which includes the aggregate and window period

### Design Notes

Working File: https://www.figma.com/file/8S8IH8mimYIwanZ6keIQ6U/Notebooks-Prototype?node-id=2460%3A12594

Prototype: https://www.figma.com/proto/8S8IH8mimYIwanZ6keIQ6U/Notebooks-Prototype?node-id=2460%3A12594&scaling=min-zoom&page-id=1915%3A0

### PR DESCRIPTION
**Pull request by drdelambre**
K, so this doesn't close all of the requirements of each of these tickets, but it is a starting point so that we can click around on it, have a conversation, and decompose some of the later design decisions encompassed in them as follow up work

Closes #1337 
Closes #1341 
Closes #1338 
Closes #1339

### CODE REVIEW

**Comment by ui-ci-bot-influx** [comment]
Metric | Score | Level
------------ | ------------- | ------------- 
performance | 55 | :red_circle:
accessibility | 96 | :green_circle:
seo | 90 | :green_circle:
pwa | null | :hankey:
best-practices | 93 | :green_circle:


[Compare your scores with master branch](https://influxdb.aws.influxdata.io/orgs/844910ece80be8bc/dashboards/073c4b95f1089000?lower=now%28%29%20-%202d&vars%5Blighthouse_score_branch%5D=1457)

**Review by Asalem1 — APPROVED** [review]

File: src/flows/components/Sidebar.tsx [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+      ],
+    },
+  ] as ControlSection[])
+    .concat(menu)
+    .map(section => {
+      const links = section.actions
+        .filter(action =>
+          typeof action.disable === 'function'
+            ? !action.disable()
+            : !action.disable
```
[the remark this answers is not in the document]
Comment (drdelambre) [reply src/flows/components/Sidebar.tsx:143]: either a boolean or function that returns a boolean (sometimes they're conditionally disabled, like export when there's no query)

File: src/flows/components/Sidebar.tsx [hunk 2]
```diff
@@ -0,0 +1,212 @@
...
+        )
+        .map(action => {
+          let title
+
+          if (typeof action.title === 'function') {
+            title = action.title()
+          } else {
+            title = action.title
```
[the remark this answers is not in the document]
Comment (drdelambre) [reply src/flows/components/Sidebar.tsx:151]: same as above

File: src/flows/context/sidebar.tsx [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+  show: _ => {},
+  hide: () => {},
+  showSub: _ => {},
+  hideSub: () => {},
+  register: (_, __) => {},
+}
+
+export const Context = createContext<ContextType>(DEFAULT_CONTEXT)
+
+export const Provider: FC = ({children}) => {
```
[the remark this answers is not in the document]
Comment (drdelambre) [reply src/flows/context/sidebar.tsx:28]: oof, this is spill over from a side project experiment where all contexts exported `Context` and `Provider` to see if it made managing large sets of contexts easier. undecided on the results, but thanks for flagging this

File: src/flows/components/panel/FlowPanelTitle.tsx [hunk 4]
```diff
@@ -38,18 +35,10 @@ const FlowPanelTitle: FC<Props> = ({id}) => {
...
         spellCheck={false}
-        maxLength={30}
       />
       <Icon glyph={IconFont.Pencil} className="flow-panel--title-icon" />
     </div>
   )
-
-  return (
-    <>
-      {sourceName}
```
[the remark this answers is not in the document]
Comment (drdelambre) [reply src/flows/components/panel/FlowPanelTitle.tsx:49]: i was amazed to see that this wasn't set anywhere?

File: src/flows/components/panel/FlowPanel.tsx [hunk 5]
```diff
@@ -1,71 +1,62 @@
...
+  const isVisible = flow.meta.get(id).visible
 
-  const moveDown = useCallback(() => {
-    if (canBeMovedDown) {
-      flow.data.move(id, index + 1)
-    }
-  }, [index, canBeMovedDown, flow.data, id])
+  const panelClassName = classnames('flow-panel', {
+    [`flow-panel__visible`]: isVisible,
+    [`flow-panel__hidden`]: !isVisible,
```
[the remark this answers is not in the document]
Comment (drdelambre) [reply src/flows/components/panel/FlowPanel.tsx:41]: because i'm not that cool

**Pull request merged**